{% from 'macros.html' import danger, info, link, span_color, section_title, modal_js, gene_redirection, fa %}
{% extends "base.html" %}
{% block title %}  {{ variant_features.gene_name[0] }}:{{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }} {% endblock %}
{% block more_head %}
	<link rel="stylesheet" href="{{ url_for('static', filename='styles/datatables.min.css') }}">
	<script src="{{ url_for('static', filename='javascript/datatables.min.js') }}" type="text/javascript"></script>
	<script src="{{ url_for('static', filename='javascript/Chart.min.js') }}" type="text/javascript"></script>
	<script src="{{ url_for('static', filename='javascript/jspdf.min.js') }}" type="text/javascript"></script>
	<script src="{{ url_for('static', filename='javascript/md_variant.js') }}?210307" type="text/javascript"></script>	
{% endblock %}
{% block content %}
	<!--left menu (hidden at first)-->
	<div class="w3-sidebar w3-bar-block w3-card w3-animate-left" id="smart_menu" style="max-height:86%;height:auto;">
		<span class="w3-bar-item w3-button w3-large w3-border-bottom w3-hover-blue" onclick="w3_close()" title="Close sidebar">X</span>
		{% set hg19_var = namespace(chr='', pos='', ref='', alt='', g_name='') %}
		{% set hg38_var = namespace(chr='', pos='', ref='', alt='', g_name='') %}
		{% for vari in variant|sort() %}
			{% if vari.genome_version == 'hg19' %}
				{# set variables #}
				{% set hg19_var.pos = vari.pos %}
				{% set hg19_var.ref = vari.pos_ref %}
				{% set hg19_var.alt = vari.pos_alt %}
				{% set hg19_var.g_name = vari.g_name %}
				{# deal with position for genome browsers #}
				{% set positions = namespace(min=vari.pos-25, max=vari.pos+25, end=vari.pos) %}
				{% if variant_features.variant_size > 1 %}
					{% set positions.end = vari.pos + variant_features.variant_size %}
					{% set positions.max = vari.pos + variant_features.variant_size +25 %}
				{% endif %}
				<button class="w3-button w3-block w3-left-align w3-large w3-hover-blue w3-border-bottom w3-blue" onclick="myAccFunc('hg19_acc', 'hg19_icon');">hg19 <i class="fa fa-caret-down" id="hg19_icon"></i></button>
				<div class="w3-show" id="hg19_acc">
					{{ link("%schr%s:%s-%s&highlight=hg19.chr%s:%s-%s&hgsid=%s"|format(urls.ucsc_hg19, vari.chr, positions.min, positions.max, vari.chr, vari.pos, positions.end, urls.ucsc_hg19_session), "hg19 UCSC")}}
					{# {{ link("%shg37&chromosome=%s&pos=%s"|format(urls.map2pdb, vari.chr, vari.pos), "hg19 Map2PDB") }} #}
					{{ link("%s&chromosome=%s&chromoStart=%s&chromoEnd=%s&x=0&y=0"|format(urls.evs, vari.chr, vari.pos, positions.end), "ESP6500") }}
					{{ link("%schr=%s&from=%s&to=%s&gts=rs%s&mk=%s:%s|rs%s"|format(urls.ncbi_1000g, vari.chr, positions.min, positions.max, variant_features.dbsnp_id, vari.pos, vari.pos, variant_features.dbsnp_id), "1000 genomes")}}
					{{ link("%svariant/%s-%s-%s-%s"|format(urls.gnomad, vari.chr, vari.pos, vari.pos_ref, vari.pos_alt), "gnomAD v2")}}
					{% if variant_features.dna_type == 'substitution' %}
						{{ link("%ssnv/GRCh37-%s/%s:%s_%s_%s"|format(urls.cadd, external_tools.CADD.version, vari.chr, vari.pos, vari.pos_ref, vari.pos_alt), "CADD")}}
					{# {% endif %}
					{% if variant_features.dna_type == 'substitution' and variant_features.start_segment_type == 'exon' %} #}
						{% if variant_features.start_segment_type == 'exon' and variant_features.p_name != '?' and variant_features.p_name != 'Met1?' %}
							{{ link("%shg19_updated.v.201904&chr=%s&pos=%s&ref=%s&alt=%s"|format(urls.intervar, vari.chr, vari.pos, vari.pos_ref, vari.pos_alt), "InterVar")}}
						{% endif %}
					{% endif %}
					{#{{ link("http://www.ncbi.nlm.nih.gov/clinvar?term='%s.%s:c.%s' [Variant name]"|format(variant_features.gene_name[1], variant_features.nm_version, variant_features.c_name), "Clinvar")}}#}
					{% if variant_features.so_accession != None %}
						{{ link("%s?term=(((%s[Base Position for Assembly GRCh37]) AND %s[Chromosome] AND \"so %s\"[Molecular consequence])) OR ((%s):c.%s [Variant name])"|format(urls.ncbi_clinvar, vari.pos, vari.chr, variant_features.so_accession, variant_features.gene_name[0], variant_features.c_name), "Clinvar search")}}
					{% else %}
						{{ link("%s?term=(((%s[Base Position for Assembly GRCh37]) AND %s[Chromosome])) OR ((%s):c.%s [Variant name])"|format(urls.ncbi_clinvar, vari.pos, vari.chr, variant_features.gene_name[0], variant_features.c_name), "Clinvar search")}}
					{% endif %}
					{{ link("%srs%s"|format(urls.ncbi_dbsnp, variant_features.dbsnp_id), "dbSNP")}}
					{% if variant_features.dbsnp_id != None %}
						{{ link("%srs%s"|format(urls.regulomedb_hg19_search, variant_features.dbsnp_id), "RegulomeDB") }}
					{% endif %}
				{#{{ link("https://varsome.com/variant/hg19/%s.%s:c.%s"|format(variant_features.gene_name[1], variant_features.nm_version, variant_features.c_name), "VarSome") }}#}
				</div>
			{% elif vari.genome_version == 'hg38' %}
				{# set variables #}
				{% set hg38_var.pos = vari.pos %}
				{% set hg38_var.ref = vari.pos_ref %}
				{% set hg38_var.alt = vari.pos_alt %}
				{% set hg38_var.g_name = vari.g_name %}
				{# deal with position for genome browsers #}
				{% set positions = namespace(min=vari.pos-25, max=vari.pos+25, end=vari.pos) %}
				{% if variant_features.variant_size > 1 %}
					{% set positions.end = vari.pos + variant_features.variant_size %}
					{% set positions.max = vari.pos + variant_features.variant_size +25 %}
				{% endif %}
				<button class="w3-button w3-block w3-left-align w3-large w3-hover-blue w3-border-bottom w3-blue" onclick="myAccFunc('hg38_acc', 'hg38_icon');">hg38 <i class="fa fa-caret-down" id="hg38_icon"></i></button>
				<div class="w3-show" id="hg38_acc">
					<span style="display:none;" id="mupit_url">{{ urls.mupit }}</span>
					<span style="display:none;" id="mupit_link">{{ link("%s?gm=chr%s:%s"|format(urls.mupit, vari.chr, vari.pos), "MuPIT") }}</span>
					{{ link("%schr%s:%s-%s&highlight=hg38.chr%s:%s-%s&hgsid=%s"|format(urls.ucsc_hg38, vari.chr, positions.min, positions.max, vari.chr, vari.pos, positions.end, urls.ucsc_hg38_session), "hg38 UCSC") }}
					{# {{ link("%shg38&chromosome=%s&pos=%s"|format(urls.map2pdb, vari.chr, vari.pos), "hg38 Map2PDB") }} #}
					{{ link("%svariant/%s-%s-%s-%s?dataset=gnomad_r3"|format(urls.gnomad, vari.chr, vari.pos, vari.pos_ref, vari.pos_alt), "gnomAD v3")}}
					{% if variant_features.dna_type == 'substitution' %}
						{{ link("%ssnv/GRCh38-%s/%s:%s_%s_%s"|format(urls.cadd, external_tools.CADD.version, vari.chr, vari.pos, vari.pos_ref, vari.pos_alt), "CADD")}}
						{% if variant_features.start_segment_type == 'exon' %}
							{{ link("%sresults.php?refSeq=>%s(%s)-%s%s\n%s&variation=c.%s&c=%s"|format(urls.hexosplice, variant_features.gene_name[1], variant_features.gene_name[0], variant_features.start_segment_type|capitalize, variant_features.start_segment_number, annot.exon_sequence, variant_features.c_name, annot.exon_first_nt_cdna_position), "HexoSplice") }}
						{% else %}
							{{ link("%sresults.php?refSeq=>%s(%s)-%s%s(51bp)\n%s&variation=26:%s"|format(urls.hexosplice, variant_features.gene_name[1], variant_features.gene_name[0], variant_features.start_segment_type|capitalize, variant_features.start_segment_number, variant_features.wt_seq|replace(' ', ''), annot.substitution_nature), "HexoSplice") }}
						{% endif %}
					{% endif %}
				</div>
			{% endif %}
		{% endfor %}
		{{ link(url_for('md.gene', gene_name=variant_features.name[0]), "MD %s"|format(variant_features.name[0])) }}
		{% if g.user %}
			{% if favourite != True %}
				<span class="w3-bar-item w3-button w3-large w3-hover-blue w3-border-bottom w3-center" id="favour_span" name='mark' onclick="favourite('{{ variant_features.var_id }}', '{{ url_for('ajax.favourite') }}', '{{ csrf_token() }}');" title="Mark as favourite"><i id="favour" class="fa fa-star"></i>
</span>
			{% else %}
				<span class="w3-bar-item w3-button w3-large w3-hover-blue w3-border-bottom w3-center" id="favour_span" name='unmark' onclick="favourite('{{ variant_features.var_id }}', '{{ url_for('ajax.favourite') }}', '{{ csrf_token() }}');" title="Unmark as favourite"><i id="favour" class="fa fa-star-o"></i>
</span>
			{% endif %}
		{% endif %}
	</div>
	<div class="w3-bar w3-light-grey" style="width:100%;z-index:1300;position:fixed;top:50px;" id="page_menu">
		<a href="#nomenclatures" class="w3-bar-item w3-button w3-mobile">Nomenclatures</a>
		<a href="#positions" class="w3-bar-item w3-button w3-mobile">Positions</a>
		<a href="#frequencies" class="w3-bar-item w3-button w3-mobile">Frequencies</a>
		<a href="#predictions" class="w3-bar-item w3-button w3-mobile">Predictions</a>
		{% if variant_features.dna_type == 'substitution' or annot.spliceai_DS_AG is defined %}
			<a href="#splicing" class="w3-bar-item w3-button w3-mobile">Splicing</a>
		{% endif %}
		{% if variant_features.prot_type == 'missense' %}
			<a href="#missense" class="w3-bar-item w3-button w3-mobile">Missense</a>
		{% endif %}
		{% if variant_features.dna_type == 'substitution' and annot.miranda_cat is defined %}
			<a href="#dbmts" class="w3-bar-item w3-button w3-mobile">miRNA target sites</a>
		{% endif %}
		<a href="#class" class="w3-bar-item w3-button w3-mobile">Classification</a>
		<a href="#admin" class="w3-bar-item w3-button w3-mobile">Admin info</a>
	</div>
	{#{{ gene_redirection(variant_features.name[0], urls) }}
	{{ modal_js('gene_modal') }}#}
	<div id="global_content" style="margin-left:15%">
		<!--first section-->
		<div id="mobile_var_name" class="w3-center w3-blue" style="display:none;">{{ "%s:%s.%s:c.%s"|format(variant_features.gene_name[0], variant_features.gene_name[1], variant_features.nm_version, variant_features.c_name) }}</div>
		<hr />
		{{ section_title('nomenclatures', 'Nomenclatures') }}
		<div class="w3-container w3-responsive">
			<table class="w3-table w3-centered display compact" style="width:90%" id="nomenclature_table">
				<thead>
					<tr>
						<th class="w3-left-align">Features</th>
						<th class="w3-left-align" width="40%">Values</th>
						<th class="w3-left-align">Descriptions</th>
					</tr>
				</thead>
				<tbody>
					<!--HGVS DNA on transcript-->
					<span style="display:none;" id="gene_name">{{ variant_features.gene_name[0] }}</span>
					<tr>
						<td class="w3-left-align">HGVS DNA on transcript:</td>
						<td class="w3-left-align" id="nm_var"><a href="{{ urls.ncbi_nuccore }}{{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}" target="_blank" title="open NCBI page for {{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}"><span id="nm_acc">{{ variant_features.gene_name[1] }}</span>.{{ variant_features.nm_version }}</a>:<span id="c_name">c.{{ variant_features.c_name }}</span></td>
						<td class="w3-left-align"><em class="w3-small">HGVS full nomenclature at DNA level on transcript</em></td>
					</tr>
					<!--<a target="_blank" title="Check Mutalyzer for the variant" href="{{ urls.mutalyzer_name_checker }}{{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }}">-->
					<!--IVS name-->
					{% if variant_features.start_segment_type == 'intron' %}
						<tr>
							<td class="w3-left-align">IVS name:</td>
							<td class="w3-left-align">{{ variant_features.ivs_name }}</td>
							<td class="w3-left-align"><em class="w3-small">IVS nomenclature</em></td>
						</tr>
					{% endif %}
					<!--HGVS Protein-->
					<tr>
						<td class="w3-left-align">HGVS Protein:</td>
						<td class="w3-left-align"><a href="{{ urls.ncbi_prot }}{{ variant_features.np }}" target="_blank" title="open NCBI page for {{ variant_features.np }}">{{ variant_features.np }}</a>:<span id="hgvs_p_name">p.({{ variant_features.p_name }})</span></td>
						<td class="w3-left-align"><em class="w3-small">HGVS full nomenclature at protein level</em></td>
					</tr>
					<!--HGVS genomic (gene)-->
					{% if variant_features.ng != 'NG_000000.0' and variant_features.ng_name != None %}
						<tr>
							<td class="w3-left-align">HGVS genomic (gene):</td>
							<td class="w3-left-align"><a href="{{ urls.ncbi_nuccore }}{{ variant_features.ng }}" target="_blank" title="open NCBI page for {{ variant_features.ng }}">{{ variant_features.ng }}</a>:g.{{ variant_features.ng_name }}</td>
							<td class="w3-left-align"><em class="w3-small">HGVS full nomenclature at genomic level (gene centered)</em></td>
						</tr>
					{% endif %}
					<!--HGVS genomic (hg19)-->
					<tr>
						<td class="w3-left-align">HGVS genomic (hg19):</td>
						<td class="w3-left-align">chr{{ variant_features.chr }}:g.<span id="hg19_g_name">{{ hg19_var.g_name }}</span></td>
						<td class="w3-left-align"><em class="w3-small">HGVS full nomenclature at genomic level (hg19)</em></td>
					</tr>
					<!--HGVS genomic strict (hg19)-->
					<tr>
						<td class="w3-left-align">HGVS strict genomic (hg19):</td>
						<td class="w3-left-align">{{ annot.ncbi_chr_hg19 }}:g.{{ hg19_var.g_name }}</td>
						<td class="w3-left-align"><em class="w3-small">HGVS full strict nomenclature at genomic level (hg19)</em></td>
					</tr>
					<!--pseudo VCF (hg19)-->
					<tr>
						<td class="w3-left-align">pseudo VCF (<span id="genome_19">hg19</span>):</td>
						<td class="w3-left-align"><a href="{{ urls.variant_validator }}{{ variant_features.chr }}-{{ hg19_var.pos }}-{{ hg19_var.ref }}-{{ hg19_var.alt }}&primary_assembly=GRCh37&alignment=splign" target="_blank" title="open VariantValidator"><span id="chrom_19">{{ variant_features.chr }}</span>-<span id="pos_19">{{ hg19_var.pos }}</span>-<span id="ref_19">{{ hg19_var.ref }}</span>-<span id="alt_19">{{ hg19_var.alt }}</span></a></td>
						<td class="w3-left-align"><em class="w3-small">chr-pos-ref-alt (hg19)</em></td>
					</tr>
					<!--HGVS genomic (hg38)-->
					<tr>
						<td class="w3-left-align">HGVS genomic (hg38):</td>
						<td class="w3-left-align">chr{{ variant_features.chr }}:g.{{ hg38_var.g_name }}</td>
						<td class="w3-left-align"><em class="w3-small">HGVS full nomenclature at genomic level (hg38)</em></td>
					</tr>
					<!--HGVS genomic strict (hg38)-->
					<tr>
						<td class="w3-left-align">HGVS strict genomic (hg38):</td>
						<td class="w3-left-align">{{ annot.ncbi_chr_hg38 }}:g.{{ hg38_var.g_name }}</td>
						<td class="w3-left-align"><em class="w3-small">HGVS full strict nomenclature at genomic level (hg38)</em></td>
					</tr>
					<!--pseudo VCF (hg38)-->
					<tr>
						<td class="w3-left-align">pseudo VCF (<span id="genome_38">hg38</span>):</td>
						<td class="w3-left-align"><a href="{{ urls.variant_validator }}{{ variant_features.chr }}-{{ hg38_var.pos }}-{{ hg38_var.ref }}-{{ hg38_var.alt }}&primary_assembly=GRCh38&alignment=splign" target="_blank" title="open VariantValidator"><span id="chrom_38">{{ variant_features.chr }}</span>-<span id="pos_38">{{ hg38_var.pos }}</span>-<span id="ref_38">{{ hg38_var.ref }}</span>-<span id="alt_38">{{ hg38_var.alt }}</span></a></td>
						<td class="w3-left-align"><em class="w3-small">chr-pos-ref-alt (hg38)</em></td>
					</tr>
				</tbody>
			</table><br/><br/>
			<!--External links-->
			<div class="w3-row-padding">
				<!--Mutalyzer-->
				<div class="w3-col m4">
					<!--<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" onclick="window.open('{{ urls.mutalyzer_name_checker }}{{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }}', '_blank');" title="Mutalyzer for {{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }}" style="width:100%">Mutalyzer</span>-->
					<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" onclick="window.open('{{ urls.mutalyzer_position_converter }}GRCh38&description=chr{{ variant_features.chr }}:g.{{ hg38_var.g_name }}', '_blank');" title="Mutalyzer other possible mappings for the variant" style="width:100%">Mutalyzer alt mappings</span>
				</div>
				<!--VariantValidator-->
				<div class="w3-col m4">
					<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" onclick="window.open('{{ urls.variant_validator }}{{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.' + encodeURIComponent('{{ variant_features.c_name }}') + '&primary_assembly=GRCh38&alignment=splign', '_blank');" title="VariantValidator for {{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }} (hg38)" style="width:100%">VariantValidator</span>
				</div>
				<!--HGVS-->
				<div class="w3-col m4">
					<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" onclick="window.open('{{ urls.hgvs }}', '_blank')" title="HGVS sequence variant nomenclature web site" style="width:100%">HGVS website</span>
				</div>
			</div>
		</div>
		<!--second section-->
		<hr />
		{{ section_title('positions', 'Positions') }}
		<div class="w3-responsive w3-container">
			<!--Position in transcript/protein-->
			<table class="w3-table w3-centered display compact" style="width:90%" id="position_table">
				<thead>
					<tr>
						<th class="w3-left-align">Features</th>
						<th class="w3-left-align" width="40%">Values</th>
						<th class="w3-left-align">Descriptions</th>
					</tr>
				</thead>
				<tbody>
					<tr><!--Position in transcript-->
						<td class="w3-left-align">Position in transcript:</td>
						{% if variant_features.start_segment_type == variant_features.end_segment_type and variant_features.start_segment_number == variant_features.end_segment_number %}
							<td class="w3-left-align">{{ variant_features.start_segment_type|capitalize }} {{ variant_features.start_segment_number }}</td>
						{% else %}
							<td class="w3-left-align">{{ variant_features.start_segment_type|capitalize }} {{ variant_features.start_segment_number }} - {{ variant_features.end_segment_type|capitalize }} {{ variant_features.end_segment_number }}</td>
						{% endif %}
						<td class="w3-left-align"><em class="w3-small">Exon/intron position in {{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}</em></td>
					</tr><!--position in exon  and variant_features.c_name|match('^[^\*-]')-->
					{% if variant_features.start_segment_type == 'exon' and variant_features.start_segment_type == variant_features.end_segment_type and variant_features.start_segment_number == variant_features.end_segment_number %}
						{% if pos_splice is not none %}
							<tr>
								<td class="w3-left-align">Position / splice site</td>
								<td class="w3-left-align">{{ annot.nearest_site_dist }} bp from {{ annot.nearest_site_type }}</td>
								<td class="w3-left-align"><em class="w3-small">Position relative to the nearest splice site</em></td>
							</tr>
						{% endif %}
						{% if variant_features.prot_type != 'unknown' %}
							{% if aa_pos[0]|int <= variant_features.prot_size|int %}
								<!-- in case prot size is wrong-->
								<tr><!--protein-->
									<td class="w3-left-align">Position / protein</td>
									<td class="w3-left-align">
										<span>{{ aa_pos[0] }}</span>
										{% if aa_pos[0] != aa_pos[1] %}
											<span> - {{ aa_pos[1] }}</span>
										{%endif %}
									 <span> / {{ variant_features.prot_size }}</span>
									</td>
									<td class="w3-left-align"><em class="w3-small">Position relative to the protein</em>
								</tr><!--protein domain-->
							{% endif %}
							<tr>
								<td class="w3-left-align">Position / domain</td>
								{% if protein_domain[0] is defined %}
									<td class="w3-left-align">
									{% for dom in protein_domain  %}
										{{ dom.name }} ({{ dom.aa_start }} - {{ dom.aa_end }})&nbsp;&nbsp;
									{% endfor %}
									</td>
								{% else %}
									<td class="w3-left-align">Not in a UNIPROT defined domain</td>
								{% endif %}
								<td class="w3-left-align"><em class="w3-small">Position in a protein domain (UNIPROT: <a href="{{ urls.uniprot_id }}{{ variant_features.uniprot_id }}" target="_blank" title="Visit UNIPROT {{ variant_features.uniprot_id }}">{{ variant_features.uniprot_id }})</a></em></td>
							</tr>
						{% endif %}
						{% if annot.metadome_dn_ds is defined %}
							<td class="w3-left-align">Position tolerance</td>
							<td class="w3-left-align">{{ annot.metadome_dn_ds }} : {{ span_color(annot.metadome_effect, annot.metadome_color) }} - <a href = "https://stuart.radboudumc.nl/metadome/dashboard/transcript/{{ annot.metadome_transcript }}/" target="_blank">Transcript view</a></td>
							<td class="w3-left-align"><em class="w3-small"><a href="{{ urls.metadome }}" target="_blank" title="Visit Metadome">MetaDome</a> score, the closer to 0, the more intolerant to variation</em></td>
						{% endif %}
					{% endif %}
					<tr>
						<td class="w3-left-align">Wild type sequence</td>
						<td class="w3-left-align txt"><textarea id="wt_seq" rows="3" cols="30" readonly>{{ variant_features.wt_seq }}</textarea></td>
						<td class="w3-left-align"><em class="w3-small">Wild type DNA sequence +/- 25 bp&nbsp;&nbsp;&nbsp;<button onclick="copy_text('wt_seq');" title="copy wild type sequence to clipboard"><i class="fa fa-clipboard" aria-hidden="true"></i></button></em></td>
					</tr>
					<tr>
						<td class="w3-left-align">Mutant sequence</td>
						<td class="w3-left-align txt"><textarea id="mt_seq" rows="3" cols="30" readonly>{{ variant_features.mt_seq }}</textarea></td>
						<td class="w3-left-align"><em class="w3-small">Mutant type DNA sequence +/- 25 bp&nbsp;&nbsp;&nbsp;<button onclick="copy_text('mt_seq');" title="copy mutant sequence to clipboard"><i class="fa fa-clipboard" aria-hidden="true"></i></button></em></td>
					</tr>
				</tbody>
			</table><br/><br/>
			<div class="w3-row-padding">
				<div class="w3-col m4">
					<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" style="width:100%" id ="litvar_data">
						Collecting LitVar data...<i class="fa fa-spinner w3-spin" style="font-size:24px"></i>
					</span>
				</div>
				<div class="w3-col m4" id="defgen_hg19">
					<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" title="generates a DEFGEN compliant csv file for the variant" style="width:100%" id="defgen_btn_hg19" onclick="defgen_export('hg19', '{{ variant_features.var_id }}', '{{ url_for('ajax.defgen') }}', '{{ csrf_token() }}');">Export to DEFGEN (hg19)</span>
					<div class="w3-modal" id="defgen_modal_hg19" style="z-index:2002;"></div>
					<!--<span id="vf_id" style="display:none;">{{ variant_features.id }}</span>-->
				</div>
				<div class="w3-col m4" id="defgen_hg38">
					<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" title="generates a DEFGEN compliant csv file for the variant" style="width:100%" id="defgen_btn_hg38" onclick="defgen_export('hg38', '{{ variant_features.var_id }}', '{{ url_for('ajax.defgen') }}', '{{ csrf_token() }}');">Export to DEFGEN (hg38)</span>
					<div class="w3-modal" id="defgen_modal_hg38" style="z-index:2002;"></div>
				</div>
			</div>
		</div>
		<!--third section-->
		<hr />
		{{ section_title('frequencies', 'Population Frequencies and Databases') }}
		<div class="w3-responsive w3-container">
			<!--Position in transcript/protein-->
			<table class="w3-table w3-centered display compact" style="width:90%" id="population_table">
				<thead>
					<tr>
						<th class="w3-left-align">Features</th>
						<th class="w3-left-align" width="60%">Values</th>
						<th class="w3-left-align">Descriptions</th>
					</tr>
				</thead>
				<tbody>
					<tr><!--gnomAD exome-->
						<td class="w3-left-align"><a href={{ "%svariant/%s-%s-%s-%s?dataset=gnomad_r2_1"|format(urls.gnomad, variant_features.chr, hg19_var.pos, hg19_var.ref, hg19_var.alt) }} target="_blank" title="Visit gnomAD for this variant">gnomAD</a> exome:</td>
						<td class="w3-left-align">{{ annot.gnomad_exome_all }}</td>
						<td class="w3-left-align"><em class="w3-small"> v2.0.1 Exomes global MAF</em></td>
					</tr>
					<tr><!--gnomAD genome-->
						<td class="w3-left-align"><a href={{ "%svariant/%s-%s-%s-%s?dataset=gnomad_r2_1"|format(urls.gnomad, variant_features.chr, hg19_var.pos, hg19_var.ref, hg19_var.alt) }} target="_blank" title="Visit gnomAD for this variant">gnomAD</a> genome:</td>
						<td class="w3-left-align">{{ annot.gnomad_genome_all }}</td>
						<td class="w3-left-align"><em class="w3-small"> v2.0.1 Genomes global MAF</em></td>
					</tr>
					<!--<tr>--><!--gnomAD v3-->
						<!--<td class="w3-left-align"><a href={{ "%svariant/%s-%s-%s-%s?dataset=gnomad_r3"|format(urls.gnomad, variant_features.chr, hg38_var.pos, hg38_var.ref, hg38_var.alt) }} target="_blank" title="Visit gnomAD for this variant">gnomAD</a> v3:</td>
						<td class="w3-left-align">{{ annot.gnomadv3 }}</td>
						<td class="w3-left-align"><em class="w3-small"> v3 Genomes global MAF</em></td>-->
					<!--</tr>-->
					<tr><!--gnomAD v3-->
						<td class="w3-left-align"><a href={{ "%svariant/%s-%s-%s-%s?dataset=gnomad_r3"|format(urls.gnomad, variant_features.chr, hg38_var.pos, hg38_var.ref, hg38_var.alt) }} target="_blank" title="Visit gnomAD for this variant">gnomAD</a> v3:</td>
						<td class="w3-left-align">{{ annot.gnomadv3 }}</td>
						<td class="w3-left-align"><em class="w3-small"> v3 Genomes global MAF</em></td>
					</tr>
					<tr><!--dbSNP id-->
						<td class="w3-left-align">dbSNP rsid:</td>
						<td class="w3-left-align">
							{% if variant_features.dbsnp_id != None %}
								<a href="{{ urls.ncbi_dbsnp }}rs{{ variant_features.dbsnp_id }}" target="_blank" title="open dbSNP" id="dbsnp_id">rs{{ variant_features.dbsnp_id }}</a>
							{% else %}
								No match in dbSNP {{ external_tools['dbSNP']['version'] }}
							{% endif %}
						</td>
						<td class="w3-left-align"><em class="w3-small">Identifier for NCBI dbSNP</em></td>
					</tr>
					<tr><!--clinvar-->
						<td class="w3-left-align">Clinvar:</td>
						<td class="w3-left-align">
							{% if annot.clinvar_id is defined %}
								<a href={{ "%s/variation/%s/"|format(urls.ncbi_clinvar, annot.clinvar_id) }}  target="_blank" title="Visit Clinvar for this variant">{{ annot.clinsig }}</a>
								{% if annot.clinsigconf is defined %}
									({{ annot.clinsigconf }})
								{% endif %}
							{% else %}
								 {{ annot.clinsig }}
							{%endif %}
						</td>
						<td class="w3-left-align"><em class="w3-small">
							{% if annot.clinvar_id is defined %}

							{% else %}
								<a href="{{ urls.ncbi_clinvar }}" target="_blank" title="Visit Clinvar">
							{%endif %}
							Clinvar</a> interpretation</em>
						</td>
					</tr>
					{% if variant_features.dna_type == 'substitution' and variant_features.start_segment_type == 'exon' and variant_features.p_name != '?' and variant_features.p_name != 'Met1?' %}
						<tr id="intervar_tr" onmouseout="$('#acmg_info').text('');"><!--intervar-->
							<td class="w3-left-align"><a href={{ "%shg19_updated.v.201904&chr=%s&pos=%s&ref=%s&alt=%s"|format(urls.intervar, variant_features.chr, hg19_var.pos, hg19_var.ref, hg19_var.alt) }} target="_blank" title="Check and adjust ACMG class at intervar">Intervar</a>:</td>
							<td class="w3-left-align">
								<span  id='intervar_data'>Asking Intervar...  <i class="fa fa-spinner w3-spin" style="font-size:24px"></i></span>
							</td>
							<td class="w3-left-align"><em class="w3-small">Semi-automated ACMG classification - click on the intervar link to adjust<br /> - passing the mouse over a criterion will display the definition</em>
							</td>
						</tr>
					{% endif %}
					<tr><!--LOVD matches-->
						<td class="w3-left-align" id="lovd_feature">LOVD matches:</td>
						<td class="w3-left-align">
							<span  id='lovd_data'>Searching LOVD...  <i class="fa fa-spinner w3-spin" style="font-size:24px"></i></span>
						</td>
						<td class="w3-left-align" id="lovd_description"><em class="w3-small">LOVD match in public instances</em></td>
					</tr>
				</tbody>
			</table><br/><br/>
			<!--hidden values to trigger intervar ajax-->
			<span style="visibility:hidden" id="segment_type">{{ variant_features.start_segment_type }}</span>
			<span style="visibility:hidden" id="dna_type">{{ variant_features.dna_type }}</span>
		</div>
		<!--fourth section-->
		<hr />
		{{ section_title('predictions', 'Overall predictions') }}
		<div class="w3-responsive w3-container">
			<table class="w3-table w3-centered display compact" style="width:90%" id="prediction_table">
				<thead>
					<tr>
						<th class="w3-left-align">Features</th>
						<th class="w3-left-align" width="40%">Values</th>
						<th class="w3-left-align">Descriptions</th>
					</tr>
				</thead>
				<tbody>
					{% if g.user['academic'] or not g.user %}
						<tr><!--CADD raw-->
							<!--https://cadd.gs.washington.edu/snv/GRCh37-v1.6_inclAnno/22:43451447_C_A-->
							{# {% if variant_features.dna_type == 'substitution' %}
								<td class="w3-left-align"><a href="{{ urls.cadd }}snv/GRCh38-{{ external_tools.CADD.version }}/{{ variant_features.chr }}:{{ hg38_var.pos }}_{{ hg38_var.ref }}_{{ hg38_var.alt }}" target="_blank" title="Visit CADD {{ external_tools.CADD.version }} for this variant">CADD</a> raw:</td>
							{% else %} #}
							<td class="w3-left-align"><a href="{{ urls.cadd }}" target="_blank" title="Visit CADD">CADD</a> raw:</td>
							{#{% endif %}#}
							{% if annot.cadd is not defined %}
								<td class="w3-left-align">{{ annot.cadd_raw }}</td>
							{% else %}
								<td class="w3-left-align">{{ annot.cadd }}</td>
							{% endif %}
							<td class="w3-left-align"><em class="w3-small">[-6.41;35.5] The higher the less likely to be observed</em></td>
						</tr>
						<tr><!--CADD phred-->
							{#{% if variant_features.dna_type == 'substitution' %}
								<td class="w3-left-align"><a href="{{ urls.cadd }}snv/GRCh38-{{ external_tools.CADD.version }}/{{ variant_features.chr }}:{{ hg38_var.pos }}_{{ hg38_var.ref }}_{{ hg38_var.alt }}" target="_blank" title="Visit CADD {{ external_tools.CADD.version }} for this variant">CADD</a> phred:</td>
							{% else %}#}
							<td class="w3-left-align"><a href="{{ urls.cadd }}" target="_blank" title="Visit CADD">CADD</a> phred:</td>
							{#{% endif %}#}
							{% if annot.cadd is not defined %}
								<td class="w3-left-align">{{ annot.cadd_phred }}</td>
							{% else %}
								<td class="w3-left-align">{{ annot.cadd }}</td>
							{% endif %}
							<td class="w3-left-align"><em class="w3-small">Phred-like scaling of raw score</em></td>
						</tr>
					{% endif %}
					{% if annot.eigen is defined or annot.eigen_raw is defined %}
						<tr><!--Eigen raw-->
							<td class="w3-left-align"><a href="{{ urls.eigen }}" target="_blank" title="Visit Eigen">Eigen</a> raw:</td>
							{% if annot.eigen is not defined %}
								<td class="w3-left-align">{{ annot.eigen_raw }}</td>
							{% else %}
								<td class="w3-left-align">{{ annot.eigen }}</td>
							{% endif %}
							<td class="w3-left-align"><em class="w3-small">[-3.33;6.84] The higher the less likely to be observed</em></td>
						</tr>
						<tr><!--Eigen phred-->
							<td class="w3-left-align"><a href="{{ urls.eigen }}" target="_blank" title="Visit Eigen">Eigen</a> phred:</td>
							{% if annot.eigen is not defined %}
								<td class="w3-left-align">{{ annot.eigen_phred }}</td>
							{% else %}
								<td class="w3-left-align">{{ annot.eigen }}</td>
							{% endif %}
							<td class="w3-left-align"><em class="w3-small">Phred-like scaling of raw score</em></td>
						</tr>
					{% endif %}
					<tr><!--MPA score-->
						<td class="w3-left-align"><a href="{{ urls.mpa }}" target="_blank" title="Visit CADD">MPA</a> score:</td>
							<td class="w3-left-align">{{ span_color(annot.mpa_score, annot.mpa_color) }}</td>
						<td class="w3-left-align"><em class="w3-small">Raw score [0;10], 10: high impact</em></td>
					</tr>
					<tr><!--MPA impact-->
						<td class="w3-left-align"><a href="{{ urls.mpa }}" target="_blank" title="Visit CADD">MPA</a> impact:</td>
							<td class="w3-left-align">{{ span_color(annot.mpa_impact.capitalize(), annot.mpa_color) }}</td>
						<td class="w3-left-align"><em class="w3-small">Impact type</em></td>
					</tr>
				</tbody>
			</table><br/><br/>
		</div>
		<!--fifth section-->
		<hr />
		{{ section_title('splicing', 'Splicing predictions') }}
		<div class="w3-responsive w3-container">
			{% if (variant_features.start_segment_type == 'exon' or (variant_features.start_segment_type == 'intron' and annot.dist_from_exon < 101)) and variant_features.start_segment_type == variant_features.end_segment_type and variant_features.variant_size < 50 %}
				<!-- Exonic Splicing context -->
				<br />{{ info("The exonic splicing context of the variant including natural splice sites scores is summarized in the graph below:") }}
				<div class="w3-container w3-center">
					<canvas class="exonic_canvas" width="600" height="150" id="segment_drawing">Your web browser does not understand HTML5. In order to display the small graph, you must use a more recent one.</canvas>
				</div>
				{% autoescape false %}
					<script type="text/javascript">
						var canvas = document.getElementById("segment_drawing");
						var context = canvas.getContext("2d");
						context.fillStyle = "#000000";
						context.font = "bold 14px sans-serif";
						context.strokeStyle = "#2196F3";
						{% if annot.preceeding_segment_type != None %}
							context.fillText("{{ annot.preceeding_segment_type }} {{ annot.preceeding_segment_number }}", 100, 25);
						{% endif %}
						{% if annot.pos_exon_canvas != None %}
							context.fillText("c.{{ variant_features.c_name }}", {{ annot.pos_exon_canvas }}, 20);
						{% else %}
							context.fillText("c.{{ variant_features.c_name }}", {{ annot.pos_intron_canvas }}, 45);
						{% endif %}
						{% if annot.following_segment_type != None %}
							context.fillText("{{ annot.following_segment_type }} {{ annot.following_segment_number }}", 500, 25);
						{% endif %}
						{% if annot.preceeding_segment_number != 'UTR' and annot.preceeding_segment_number != None %}
							context.fillText("3'ss MaxEnt: {{ annot.nat3ss_score }}", 150, 125);
							context.fillText("{{ annot.nat3ss_seq }}", 100, 140);
						{% endif %}
						{% if annot.following_segment_number != 'UTR' and annot.following_segment_number != None %}
							context.fillText("5'ss MaxEnt: {{ annot.nat5ss_score }}", 400, 125);
							context.fillText("{{ annot.nat5ss_seq }}", 400, 140);
						{% endif %}
						context.moveTo(100,75);
						context.lineTo(200,75);
						context.stroke();
						{% if annot.pos_exon_canvas != None %}
							context.moveTo({{ annot.pos_exon_canvas }},20);
							context.lineTo({{ annot.pos_exon_canvas }},20+30);
						{% else %}
							context.moveTo({{ annot.pos_intron_canvas }},45);
							context.lineTo({{ annot.pos_intron_canvas }},45+30);
						{% endif %}
						context.stroke();
						context.fillStyle = "#fdf5e6"
						context.fillRect(200,50,200,50);
						context.strokeRect(200,50,200,50);
						context.fillStyle = "#000000";
						{% if variant_features.start_segment_type == 'exon' %}
							context.fillText("Exon {{ variant_features.start_segment_number }} ({{ annot.segment_size }} bp)", 250, 75);
						{% else %}
							context.fillText("Exon {{ annot.neighb_exon_number }}", 250, 75);
						{% endif %}
						context.moveTo(400,75);
						context.lineTo(500,75);
						context.stroke();
					</script>
				{% endautoescape %}
			{% endif %}
			<!-- MaxEntScan 5'ss -->
			{# {% if mes5 != None or (mes3 != None and mes3 != 'Not performed') %} #}
				{{ info('MaxEntScan scores are presented in the two following tables. Selected scores have: <ul><li>a |variation| > 15% and</li><li>a raw score for mutant or wild-type of at least 3</li></ul> The upper sequence shows the variation site in red, and the lower sequence the putative splice site considered by <a href ="http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html" target="_blank">MaxEntScan</a> (putative introns are shown as lower case and exons as upper case).') }}
			{#{% endif %}#}
			<br />
			<table class="w3-table w3-centered display compact" style="width:90%;" id="maxent5ss_table">
				<caption>MaxEntScan 5'ss scores</caption>
				<thead>
					<tr>
						<th class="w3-left-align">Wild-type sequence</th>
						<th class="w3-left-align">Score</th>
						<th class="w3-left-align">Mutant sequence</th>
						<th class="w3-left-align">Score</th>
						<th class="w3-left-align">Variation(%)</th>
					</tr>
				</thead>
				<tbody>
					{% if mes5 != None %}
						{% autoescape false %}
							{% for score in mes5 %}
								<tr>
									<td>... {{ mes5[score][5] }}...<br />... {{ mes5[score][6] }} ...</td>
									<td style="vertical-align:middle;">{{ mes5[score][1] }}</td>
									<td>... {{ mes5[score][7] }}...<br />... {{ mes5[score][8] }} ...</td>
									<td style="vertical-align:middle;">{{ mes5[score][3] }}</td>
									{% if mes5[score][4] > 0 %}
										<td style="vertical-align:middle;">+{{ mes5[score][4] }}</td>
									{% else %}
										<td style="vertical-align:middle;">{{ mes5[score][4] }}</td>
									{% endif %}
								</tr>
							{% endfor %}
						{% endautoescape %}
					{% else %}
						<tr><td></td><td></td><td>No significant MaxEnt 5'ss scores found.</td><td></td><td></td></tr>
					{% endif %}
				</tbody>
			</table><br/><br/>
			<!-- MaxEntScan 3'ss -->
			<table class="w3-table w3-centered display compact" style="width:90%" id="maxent3ss_table">
				<caption>MaxEntScan 3'ss scores</caption>
				<thead>
					<tr>
						<th class="w3-left-align">Wild-type sequence</th>
						<th class="w3-left-align">Score</th>
						<th class="w3-left-align">Mutant sequence</th>
						<th class="w3-left-align">Score</th>
						<th class="w3-left-align">Variation(%)</th>
					</tr>
				</thead>
				<tbody>
					{% if mes3 != None and mes3 != 'Not performed' %}
						{% autoescape false %}
							{% for score in mes3 %}
								<tr>
									<td>... {{ mes3[score][5] }}...<br />... {{ mes3[score][6] }} ...</td>
									<td style="vertical-align:middle;">{{ mes3[score][1] }}</td>
									<td>... {{ mes3[score][7] }}...<br />... {{ mes3[score][8] }} ...</td>
									<td style="vertical-align:middle;">{{ mes3[score][3] }}</td>
									{% if mes3[score][4] > 0 %}
										<td style="vertical-align:middle;">+{{ mes3[score][4] }}</td>
									{% else %}
										<td style="vertical-align:middle;">{{ mes3[score][4] }}</td>
									{% endif %}
								</tr>
							{% endfor %}
						{% endautoescape %}
					{% elif mes3 == 'Not performed' %}
						<tr><td></td><td></td><td>No MaxEnt 3'ss score performed (exonic variant far form 3'ss).</td><td></td><td></td></tr>
					{% else %}
						<tr><td></td><td></td><td>No significant MaxEnt 3'ss scores found.</td><td></td><td></td></tr>
					{% endif %}
				</tbody>
			</table><br/><br/>
			<!-- SPiP -->
			{{ info('<a href="%s" target="_blank">SPiP</a> results and predictions:' | format(urls.spip)) }}
			<div id="spip">{{ info('MobiDetails is running SPiP... <i class="fa fa-spinner w3-spin" style="font-size:24px"></i>') }}</div>
			<!--dbscSNV / spliceAI-->
			{{ info('This last table present dbscSNV and spliceAi scores (max distance: 50 bp), when available.') }}
			{% if variant_features.dna_type == 'substitution' or annot.spliceai_DS_AG is defined %}
				<table class="w3-table w3-centered display compact" style="width:90%" id="splicing_table">
					<thead>
						<tr>
							<th class="w3-left-align">Features</th>
							<th class="w3-left-align" width="40%">Values</th>
							<th class="w3-left-align">Descriptions</th>
						</tr>
					</thead>
					<tbody>
						{% if variant_features.dna_type == 'substitution' %}
							<tr><!--dbscSNV ADA-->
								<td class="w3-left-align"><a href="{{ urls.dbscsnv }}" target="_blank" title="Visit dbscSNV">dbscSNV</a> ADA:</td>
								<td class="w3-left-align">{{ span_color(annot.dbscsnv_ada, annot.dbscsnv_ada_color) }}</td>
								<td class="w3-left-align"><em class="w3-small">Raw score (0-1), threshold &gt; {{ thresholds.dbscsnv }} for impact</em></td>
							</tr>
							<tr><!--dbscSNV RF-->
								<td class="w3-left-align"><a href="{{ urls.dbscsnv }}" target="_blank" title="Visit dbscSNV">dbscSNV</a> RF:</td>
								<td class="w3-left-align">{{ span_color(annot.dbscsnv_rf, annot.dbscsnv_rf_color) }}</td>
								<td class="w3-left-align"><em class="w3-small">Raw score (0-1), threshold &gt; {{ thresholds.dbscsnv }} for impact</em></td>
							</tr>
						{% endif %}
						{% if annot.spliceai_DS_AG is defined %}
							<tr><!--spliceAI AG-->
								<td class="w3-left-align"><a href="{{ urls.spliceai }}" target="_blank" title="Visit spliceAI">spliceAI</a> AG:</td>
								<td class="w3-left-align">{{ span_color(annot.spliceai_DS_AG, annot.spliceai_DS_AG_color) }} ({{ annot.spliceai_DP_AG }})</td>
								<!--<td class="w3-left-align"><span style="color:{{ annot.spliceai_DS_AG_color }};">{{ annot.spliceai_DS_AG }}</span> ({{ annot.spliceai_DP_AG }})</td>-->
								<td class="w3-left-align"><em class="w3-small">Acceptor Gain, 0-1 (distance in bp), thresholds &gt; {{ "%s|%s|%s"|format(thresholds.spliceai_min, thresholds.spliceai_mid, thresholds.spliceai_max) }} for impact</em></td>
							</tr>
							<tr><!--spliceAI AL-->
								<td class="w3-left-align"><a href="{{ urls.spliceai }}" target="_blank" title="Visit spliceAI">spliceAI</a> AL:</td>
								<td class="w3-left-align">{{ span_color(annot.spliceai_DS_AL, annot.spliceai_DS_AL_color) }} ({{ annot.spliceai_DP_AL }})</td>
								<td class="w3-left-align"><em class="w3-small">Acceptor Loss, 0-1 (distance in bp), thresholds &gt; {{ "%s|%s|%s"|format(thresholds.spliceai_min, thresholds.spliceai_mid, thresholds.spliceai_max) }} for impact</em></td>
							</tr>
							<tr><!--spliceAI DG-->
								<td class="w3-left-align"><a href="{{ urls.spliceai }}" target="_blank" title="Visit spliceAI">spliceAI</a> DG:</td>
								<td class="w3-left-align">{{ span_color(annot.spliceai_DS_DG, annot.spliceai_DS_DG_color) }} ({{ annot.spliceai_DP_DG }})</td>
								<td class="w3-left-align"><em class="w3-small">Donor Gain, 0-1 (distance in bp), thresholds &gt; {{ "%s|%s|%s"|format(thresholds.spliceai_min, thresholds.spliceai_mid, thresholds.spliceai_max) }} for impact</em></td>
							</tr>
							<tr><!--spliceAI DL-->
								<td class="w3-left-align"><a href="{{ urls.spliceai }}" target="_blank" title="Visit spliceAI">spliceAI</a> DL:</td>
								<td class="w3-left-align">{{ span_color(annot.spliceai_DS_DL, annot.spliceai_DS_DL_color) }} ({{ annot.spliceai_DP_DL }})</td>
								<td class="w3-left-align"><em class="w3-small">Donor Loss, 0-1 (distance in bp), thresholds &gt; {{ "%s|%s|%s"|format(thresholds.spliceai_min, thresholds.spliceai_mid, thresholds.spliceai_max) }} for impact</em></td>
							</tr>
						{% elif annot.spliceai is defined %}
							<tr>
								<td class="w3-left-align"><a href="{{ urls.spliceai }}" target="_blank" title="Visit spliceAI">spliceAI</a>:</td>
								<td class="w3-left-align">{{ annot.spliceai }}</td>
								<td class="w3-left-align"><em class="w3-small">Donor Loss, 0-1 (distance in bp), thresholds &gt; {{ "%s|%s|%s"|format(thresholds.spliceai_min, thresholds.spliceai_mid, thresholds.spliceai_max) }} for impact</em></td>
							</tr>
						{% endif %}
					</tbody>
				</table><br/><br/>
				{% if annot.spliceai_DS_AG is defined %}
					<div class="w3-row-padding">
						<!--Splicing radar chart-->
						<div class="w3-col m4">
							<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" onclick="$('#radar_splicing_modal').show();" title="Display a radar chart of splicing scores" style="width:100%">Show radar chart</span>
						</div>
					</div>
					<div id="radar_splicing_modal" class="w3-modal w3-animate-opacity" style="display:none;z-index:2002;">
						<div class="w3-modal-content w3-card-4 w3-display-middle w3-animate-opacity w3-mobile">
							<header class="w3-container w3-blue">
								<span class="w3-button w3-display-topright w3-xlarge" onclick="$('#radar_splicing_modal').hide();">X</span>
								<br /><span class="w3-xlarge">Radar view of splicing predictors</span><br />
								<p>Values are normalised (0-1), 0 being the less damaging and 1 the most for each predictor.</p>
							</header>
							<div class="w3-container">
								<div class="chart-container" style="position: relative;">
									<canvas id="splicing_radar"><p>Radar Chart for splicing predictions</p></canvas>
								</div>
							</div>
						</div>
					</div>
					<script>
						Chart.defaults.global.defaultFontSize=16;
						var point_font_size = 16
						var wi = $(window).width();
						if (wi < 400) {
							Chart.defaults.global.defaultFontSize=12;
							point_font_size = 12;
						}
						{% autoescape false %}
							var myRadarChart = new Chart($("#splicing_radar"), {
								type: 'radar',
								data: {
									//labels: ['dbscSNV ADA', 'dbscSNV RF', 'SpliceAI Acc Gain', 'SpliceAI Acc Loss', 'SpliceAI Donor Gain', 'SpliceAI Donor Loss'],
									labels: {{ splicing_radar_labels }},
									datasets: [{
										label:'Splicing predictors',
										data: {{ splicing_radar_values }},
										fill: true,
										backgroundColor:'rgb(255, 99, 132, 0.2)',
										borderColor:'rgb(255, 99, 132)',
										pointBackgroundColor:'rgb(255, 99, 132)',
										pointBorderColor:'#fff',
										pointHoverBackgroundColor:'#fff',
										pointHoverBorderColor:'rgb(255, 99, 132)'
									}]
								},
								options:{elements:{line:{tension:0,borderWidth:3}},scale:{pointLabels:{fontSize:point_font_size},ticks:{beginAtZero:true,min:0,max:1,stepSize:0.2}},multiTooltipTemplate : "<%=label%> : <%=value%>",onAnimationComplete: "done"}
							});
						{% endautoescape %}
					</script>
				{% endif %}
			{% endif %}
		</div>
		{# {{ modal_js('radar_splicing_modal') }} #}
		<!--alt sixth section-->
		{% if variant_features.dna_type == 'substitution' and annot.dbmts is not defined and annot.miranda_cat is defined%}
			<hr />
			{{ section_title('dbmts', 'miRNA target sites predictions') }}
			<div class="w3-responsive w3-container">
				<!--dbMTS-->
				 {{ info("The table below comes form <a href='http://www.liulab.science/dbmts.html' target='_blank'>dbMTS{}</a>. It focuses on miRNA binding capabilities of the wild-type and mutant sequences and on the differences observed using 3 different algorithms, namely <a href='http://www.microrna.org/microrna/getGeneForm.do' target='_blank'>miRanda</a>, <a href='http://www.rnaiweb.com/cgi-bin/web/jump.cgi?ID=56' target='_blank'>TargetScan</a> and <a href='https://bibiserv.cebitec.uni-bielefeld.de/rnahybrid/' target='_blank'>RNAHybrid</a>. Each predictor may define three categories of functional change (see Descriptions). To correctly assess the Max Diff column, one needs to know the theoretical max differences for each tool: miRanda (216), TargetScan (11.07), RNAHybrid (60.8). Concerning best scores, for miRanda and RNAhybrid, the best score is the miRNA-target pair that has the highest prediction score. For TargetScan, the best score is the miRNA-target pair that has the largest negative prediction score.".format(external_tools['dbMTS']['version'])) }}
				<table class="w3-table w3-centered display compact" style="width:90%" id="dbmts_table">
					<thead>
						<tr>
							<th class="w3-left-align">Features</th>
							<th class="w3-left-align">miRanda</th>
							<th class="w3-left-align">TargetScan v7.0</th>
							<th class="w3-left-align">RNAhybrid</th>
							<th class="w3-left-align">Descriptions</th>
						</tr>
					</thead>
					<tbody>
						<tr><!--category-->
							<td class="w3-left-align">Category:</td>
							<td class="w3-left-align">{{ annot.miranda_cat }}</td>
							<td class="w3-left-align">{{ annot.targetscan_cat }}</td>
							<td class="w3-left-align">{{ annot.rnahybrid_cat }}</td>
							<td class="w3-left-align"><em class="w3-small">Predicted functional category of this variant according to each predictor. 'S': substitution of regulating miRNAs. 'L': loss of all miRNAs. 'G' gain.</em></td>
						</tr>
						<tr><!--rankscore-->
							<td class="w3-left-align">Rank Score:</td>
							<td class="w3-left-align">{{ annot.miranda_rankscore }}</td>
							<td class="w3-left-align">{{ annot.targetscan_rankscore }}</td>
							<td class="w3-left-align">{{ annot.rnahybrid_rankscore }}</td>
							<td class="w3-left-align"><em class="w3-small">Rank of the max_dif score among all max_dif scores within the same category [0;1]. The closer to 1 the more likely to highlight a significant change.</em></td>
						</tr>
						<tr><!--maxdiff-->
							<td class="w3-left-align">Max Diff:</td>
							<td class="w3-left-align">{{ annot.miranda_maxdiff }}</td>
							<td class="w3-left-align">{{ annot.targetscan_maxdiff }}</td>
							<td class="w3-left-align">{{ annot.rnahybrid_maxdiff }}</td>
							<td class="w3-left-align"><em class="w3-small">The absolute value of the maximum potential difference between all the scores (ref and alt).</em></td>
						</tr>
						{% autoescape false %}
							<tr><!--refbestmir-->
								<td class="w3-left-align">Ref Best Mir:</td>
								<td class="w3-left-align">{{ annot.miranda_refbestmir }}</td>
								<td class="w3-left-align">{{ annot.targetscan_refbestmir }}</td>
								<td class="w3-left-align">{{ annot.rnahybrid_refbestmir }}</td>
								<td class="w3-left-align"><em class="w3-small">miRNA corresponding to the best score of the reference (wild-type) sequence.</em></td>
							</tr>
							<tr><!--refbestscore-->
								<td class="w3-left-align">Ref Best Score:</td>
								<td class="w3-left-align">{{ annot.miranda_refbestscore }}</td>
								<td class="w3-left-align">{{ annot.targetscan_refbestscore }}</td>
								<td class="w3-left-align">{{ annot.rnahybrid_refbestscore }}</td>
								<td class="w3-left-align"><em class="w3-small">Best score for the reference sequence for each predictor.</em></td>
							</tr>
							<tr><!--altbestmir-->
								<td class="w3-left-align">Alt Best Mir:</td>
								<td class="w3-left-align">{{ annot.miranda_altbestmir }}</td>
								<td class="w3-left-align">{{ annot.targetscan_altbestmir }}</td>
								<td class="w3-left-align">{{ annot.rnahybrid_altbestmir }}</td>
								<td class="w3-left-align"><em class="w3-small">miRNA corresponding to the best score of the alternative (mutant) sequence.</em></td>
							</tr>
							<tr><!--altbestscore-->
								<td class="w3-left-align">Alt Best Score:</td>
								<td class="w3-left-align">{{ annot.miranda_altbestscore }}</td>
								<td class="w3-left-align">{{ annot.targetscan_altbestscore }}</td>
								<td class="w3-left-align">{{ annot.rnahybrid_altbestscore }}</td>
								<td class="w3-left-align"><em class="w3-small">Best score for the alternative sequence for each predictor.</em></td>
							</tr>
						{% endautoescape %}
					</tbody>
				</table>
			</div>
		{% elif annot.dbmts is defined %}
			{{ info(annot.dbmts) }}
		{% endif %}
		<!--sixth section-->
		{% if variant_features.prot_type == 'missense' and annot.dbnsfp is not defined %}
			<hr />
			{{ section_title('missense', 'Missense predictions: p.(%s)'|format(variant_features.p_name)) }}
			<div class="w3-responsive w3-container">
				<!--dbscSNV-->
				<table class="w3-table w3-centered display compact" style="width:90%" id="missense_table">
					<thead>
						<tr>
							<th class="w3-left-align">Features</th>
							<th class="w3-left-align">Values</th>
							<th class="w3-left-align">Prediction</th>
							<th class="w3-left-align">Descriptions</th>
						</tr>
					</thead>
					<tbody>
						<tr><!--SIFT-->
							<td class="w3-left-align"><a href="{{ urls.sift }}" target="_blank" title="Visit SIFT 4G">SIFT</a>:</td>
							<td class="w3-left-align">{{ span_color(annot.sift_score, annot.sift_color) }}{{ annot.sift_star }}</td>
							<td class="w3-left-align">{{ span_color(annot.sift_pred, annot.sift_color) }}</td>
							<td class="w3-left-align"><em class="w3-small">Threshold &lt; 0.05 for Damaging</em></td>
						</tr>
						{% if g.user['academic'] or not g.user %}
							<tr><!--PH2 HDIV-->
								<td class="w3-left-align"><a href="{{ urls.pph2 }}" target="_blank" title="Visit Polyphen">Polyphen 2</a> HumDiv:</td>
								<td class="w3-left-align">{{ span_color(annot.pph2_hdiv_score, annot.pph2_hdiv_color) }}{{ annot.pph2_hdiv_star }}</td>
								<td class="w3-left-align">{{ span_color(annot.pph2_hdiv_pred, annot.pph2_hdiv_color) }}</td>
								<td class="w3-left-align"><em class="w3-small">Thresholds &gt; {{ "%s|%s"|format(thresholds.pph2_hdiv_mid, thresholds.pph2_hdiv_max) }} for Possibly and Probably Damaging</em></td>
							</tr>
							<tr><!--PH2 HVAR-->
								<td class="w3-left-align"><a href="{{ urls.pph2 }}" target="_blank" title="Visit Polyphen">Polyphen 2</a> HumVar:</td>
								<td class="w3-left-align">{{ span_color(annot.pph2_hvar_score, annot.pph2_hvar_color) }}{{ annot.pph2_hvar_star }}</td>
								<td class="w3-left-align">{{ span_color(annot.pph2_hvar_pred, annot.pph2_hvar_color) }}</td>
								<td class="w3-left-align"><em class="w3-small">Thresholds &gt; {{ "%s|%s"|format(thresholds.pph2_hvar_mid, thresholds.pph2_hvar_max) }} for Possibly and Probably Damaging</em></td>
							</tr>
							<tr><!--REVEL-->
								<td class="w3-left-align"><a href="{{ urls.revel }}" target="_blank" title="Visit REVEL">REVEL</a>:</td>
								<td class="w3-left-align">{{ span_color(annot.revel_score, annot.revel_color) }}</td>
								<td class="w3-left-align">{{ span_color(annot.revel_pred, annot.revel_color) }}</td>
								<td class="w3-left-align"><em class="w3-small">Thresholds 0.2|0.5 for Benign, Uncertain, Damaging</em></td>
							</tr>
							<tr><!--ClinPred-->
								<td class="w3-left-align"><a href="{{ urls.clinpred }}" target="_blank" title="Visit ClinPred">ClinPred</a>:</td>
								<td class="w3-left-align">{{ span_color(annot.clinpred_score, annot.clinpred_color) }}{{ annot.clinpred_star }}</td>
								<td class="w3-left-align">{{ span_color(annot.clinpred_pred, annot.clinpred_color) }}</td>
								<td class="w3-left-align"><em class="w3-small">Threshold &ge; {{ thresholds.clinpred }} for Damaging</em></td>
							</tr>
						{% endif %}
						<tr><!--FATHMM-->
							<td class="w3-left-align"><a href="{{ urls.fathmm }}" target="_blank" title="Visit fathmm">Fathmm</a>:</td>
							<td class="w3-left-align">{{ span_color(annot.fathmm_score, annot.fathmm_color) }}{{ annot.fathmm_star }}</td>
							<td class="w3-left-align">{{ span_color(annot.fathmm_pred, annot.fathmm_color) }}</td>
							<td class="w3-left-align"><em class="w3-small">Threshold &le; {{ thresholds.fathmm }} for Damaging</em></td>
						</tr>
						<tr><!--Meta SVM-->
							<td class="w3-left-align"><a href="{{ urls.dbnsfp }}" target="_blank" title="Visit dbnsfp">Meta SVM</a>:</td>
							<td class="w3-left-align">{{ span_color(annot.msvm_score, annot.msvm_color) }} ({{ annot.m_rel }})</td>
							<td class="w3-left-align">{{ span_color(annot.msvm_pred, annot.msvm_color) }}</td>
							<td class="w3-left-align"><em class="w3-small">Threshold &ge; {{ thresholds['meta-svm'] }} for Damaging (reliabilty index: 0-10), 10:high</em></td>
						</tr>
						<tr><!--Meta LR-->
							<td class="w3-left-align"><a href="{{ urls.dbnsfp }}" target="_blank" title="Visit dbnsfp">Meta LR</a>:</td>
							<td class="w3-left-align">{{ span_color(annot.mlr_score, annot.mlr_color) }} ({{ annot.m_rel }})</td>
							<td class="w3-left-align">{{ span_color(annot.mlr_pred, annot.mlr_color) }}</td>
							<td class="w3-left-align"><em class="w3-small">Threshold &ge; {{ thresholds['meta-lr'] }}  for Damaging (reliabilty index: 0-10), 10:high</em></td>
						</tr>
						<tr><!--Mistic-->
							<td class="w3-left-align"><a href="{{ urls.mistic }}" target="_blank" title="Visit Mistic">Mistic</a>:</td>
							<td class="w3-left-align">{{ span_color(annot.mistic_score, annot.mistic_color) }}</td>
							<td class="w3-left-align">{{ span_color(annot.mistic_pred, annot.mistic_color) }}</td>
							<td class="w3-left-align"><em class="w3-small">Threshold &ge; {{ thresholds['mistic'] }}  for Damaging</em></td>
						</tr>
					</tbody>
				</table>
				{% if annot.sift_star == '*' or annot.pph2_hdiv_star == '*' or annot.pph2_hvar_star == '*' or annot.fathmm_star == '*' %}
					<p>*No score found for the current isoform. The reported score is the most deleterious found in other isoforms.</p>
				{% endif %}
				<br/><br/>
				<div class="w3-row-padding">
					<!--Missense radar chart-->
					<div class="w3-col m4">
						<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" onclick="$('#radar_missense_modal').show();" title="Display a radar chart of missense scores" style="width:100%">Show radar chart</span>
					</div>
				</div>
				<div id="radar_missense_modal" class="w3-modal w3-animate-opacity" style="display:none;z-index:2002;">
					<div class="w3-modal-content w3-card-4 w3-display-middle w3-animate-opacity w3-mobile">
						<header class="w3-container w3-blue">
							<span class="w3-button w3-display-topright w3-xlarge" onclick="$('#radar_missense_modal').hide();">X</span>
							<br /><span class="w3-xlarge">Radar view of missense predictors</span><br />
							<p>Values are normalised (0-1), 0 being the less damaging and 1 the most for each predictor.</p>
						</header>
						<div class="w3-container">
							<div class="chart-container" style="position: relative;">
								<canvas id="missense_radar"><p>Radar Chart for missense predictions</p></canvas>
							</div>
						</div>
					</div>
				</div>
				{# set missense predictors values to min when no values to plot radar chart #}
				{% set missense_pred = namespace(sift=annot.sift_score, pphvar=annot.pph2_hvar_score, pphdiv=annot.pph2_hdiv_score, revel=annot.revel_score, clinpred=annot.clinpred_score, fathmm=annot.fathmm_score, metasvm=annot.msvm_score, metalr=annot.mlr_score, mistic=annot.mistic_score) %}
				{% if annot.sift_pred == 'no prediction' %}
					{% set missense_pred.sift = 1 %}
				{% endif %}
				{% if annot.pph2_hvar_pred == 'no prediction' %}
					{% set missense_pred.pphvar = 0 %}
				{% endif %}
				{% if annot.pph2_hdiv_pred == 'no prediction' %}
					{% set missense_pred.pphdiv = 0 %}
				{% endif %}
				{% if annot.revel_pred == 'no prediction' %}
					{% set missense_pred.revel = 0 %}
				{% endif %}
				{% if annot.clinpred == 'no prediction' %}
					{% set missense_pred.clinpred = 0 %}
				{% endif %}
				{% if annot.fathmm_pred == 'no prediction' %}
					{% set missense_pred.fathmm = 10.64 %}
				{% endif %}
				{% if annot.msvm_pred == 'no prediction' %}
					{% set missense_pred.metasvm = -2 %}
				{% endif %}
				{% if annot.mlr_pred == 'no prediction' %}
					{% set missense_pred.metalr = 0 %}
				{% endif %}
				{% if annot.mistic == 'no prediction' %}
					{% set missense_pred.mistic = 0 %}
				{% endif %}
				<script>
					//var ctx = document.getElementById("missense_radar").getContext('2d')
					Chart.defaults.global.defaultFontSize=16;
					var point_font_size = 16
					var wi = $(window).width();
					if (wi < 400) {
						Chart.defaults.global.defaultFontSize=12;
						point_font_size = 12;
					}

					function normalizeData(val, my_array) {
						var new_val = val == 0 ? 0 : ((val - my_array[0]) / (my_array[1] - my_array[0])).toFixed(2);
						return new_val;
					}
					var myRadarChart = new Chart($("#missense_radar"), {
						type: 'radar',
						data: {
							labels: ['SIFT', {% if g.user['academic'] or not g.user %}'Polyphen 2 HDIV', 'Polyphen 2 HVAR', 'REVEL', 'ClinPred', {% endif %}'FATHMM', 'MetaSVM', 'MetaLR', 'Mistic'],
							datasets: [{
								label:'Missense predictors',
								data: [1-{{ missense_pred.sift }}, {% if g.user['academic'] or not g.user %} {{ missense_pred.pphdiv }}, {{ missense_pred.pphvar }}, {{ missense_pred.revel }}, {{ missense_pred.clinpred }}, {% endif %}1-normalizeData({{ missense_pred.fathmm }}, [-16.13,10.64]), normalizeData({{ missense_pred.metasvm }}, [-2,3]), {{ missense_pred.metalr }}, {{ missense_pred.mistic }}],
								fill: true,
								backgroundColor:'rgb(255, 99, 132, 0.2)',
								borderColor:'rgb(255, 99, 132)',
								pointBackgroundColor:'rgb(255, 99, 132)',
								pointBorderColor:'#fff',
								pointHoverBackgroundColor:'#fff',
								pointHoverBorderColor:'rgb(255, 99, 132)'
							}]
						},
						options:{elements:{line:{tension:0,borderWidth:3}},scale:{pointLabels:{fontSize:point_font_size},ticks:{beginAtZero:true,min:0,max:1,stepSize:0.2}},multiTooltipTemplate : "<%=label%> : <%=value%>",onAnimationComplete: "done"}
					});
				</script>
				{% if g.user %}
					{{ modal_js('pubmed', 'radar_splicing_modal', 'radar_missense_modal', 'defgen_modal_hg19', 'defgen_modal_hg38', 'modify_class_modal') }}
				{% else %}
					{{ modal_js('pubmed', 'radar_splicing_modal', 'radar_missense_modal', 'defgen_modal_hg19', 'defgen_modal_hg38') }}
				{% endif %}
			</div>


		{% elif annot.dbnsfp is defined %}
			{{ info(annot.dbnsfp) }}
			{% if g.user %}
				{{ modal_js('pubmed', 'radar_splicing_modal', 'defgen_modal_hg19', 'defgen_modal_hg38', 'modify_class_modal') }}
			{% else %}
				{{ modal_js('pubmed', 'radar_splicing_modal', 'defgen_modal_hg19', 'defgen_modal_hg38') }}
			{% endif %}
		{% elif annot.spliceai_DS_AG is defined  %}
			{% if g.user %}
				{{ modal_js('pubmed', 'radar_splicing_modal', 'defgen_modal_hg19', 'defgen_modal_hg38', 'modify_class_modal') }}
			{% else %}
				{{ modal_js('pubmed', 'radar_splicing_modal', 'defgen_modal_hg19', 'defgen_modal_hg38') }}
			{% endif %}
		{% else %}
			{% if g.user %}
				{{ modal_js('pubmed', 'defgen_modal_hg19', 'defgen_modal_hg38', 'modify_class_modal') }}
			{% else %}
				{{ modal_js('pubmed', 'defgen_modal_hg19', 'defgen_modal_hg38') }}
			{% endif %}
		{% endif %}
		<!--seventh section-->
		<hr />
		{{ section_title('class', 'Classification History') }}
		<div class="w3-responsive w3-container">
			{% if g.user %}
				<div class="w3-margin w3-panel w3-sand w3-leftbar w3-display-container">
					<p>By assigning an ACMG class to this variant, you also contribute to the international knowledge on DNA variants, as we will submit the variant and the classification to the <a href="https://databases.lovd.nl/shared/genes/{{ variant_features.name[0] }}" target="_blank">Global Variome Shared LOVD</a> (Read <a href="{{ url_for('md.about') }}#data_sharing" target="_blank">more</a>).</p>
					<p>
						{% if g.user['lovd_export'] == true %}
							Your LOVD export status is currently <span class="w3-text-green">On</span> - This means the data will be exported.
						{% else %}
							Your LOVD export status is currently <span class="w3-text-red">Off</span> - This means the data will not be exported.
						{% endif %}
						<br /> You can modify it at your <a href="{{ url_for('auth.profile', mobiuser_id=0) }}" target="_blank">profile</a> page.
					</p>
				</div>
			{% endif %}
			<!--ACMG Classifications-->
			<table class="w3-table w3-centered display compact" style="width:90%" id="class_table">
				<thead>
					<tr>
						<th class="w3-left-align">User</th>
						<th class="w3-left-align">Date</th>
						<th class="w3-left-align">ACMG Classification</th>
						<th class="w3-left-align">Comments</th>
					</tr>
				</thead>
				<tbody>
				{% if class_history == None %}
					<tr id="no_class">
						<td></td>
						<td></td>
						<td>No classification defined yet for this variant</td>
						<td></td>
					</tr>
				{% else %}
					{% for classif in class_history %}
						<tr id="{{ classif.id }}-{{ classif.acmg_class }}-{{ variant_features.var_id }}"><!--classification infos-->
							{% if g.user %}
								{% if g.user['id'] == classif.id %}
									<td class="w3-left-align"><a href="{{ url_for('auth.profile', mobiuser_id=0) }}" target="_blank" title="Visit your profile">{{ classif.username }}</a></td>
								{% elif classif.username != 'mobidetails' %}
									<td class="w3-left-align">
										<a href="{{ url_for('auth.profile', mobiuser_id=classif.id) }}" target="_blank" title="See {{ classif.username }} profile">{{ classif.username }}</a>
										{% if classif.email_pref == true %}
											&nbsp;&nbsp;<i class="w3-xlarge fa fa-envelope-o" onclick="$('#receiver_id').val('{{ classif.id }}');$('#receiver_username').html('{{ classif.username }}');$('#message_modal').show();" title="Send {{ classif.username }} a message via mobidetails" style="cursor:pointer;"></i>
										{% endif %}
									</td>
								{% else %}
									<td class="w3-left-align">{{ classif.username }}</td>
								{% endif %}
							{% else %}
								<td class="w3-left-align">{{ classif.username }}</td>
							{% endif %}
							<td class="w3-left-align">{{ classif.class_date }}</td>
							<td class="w3-left-align"><span style="color:{{ classif.html_code }};">Class {{ classif.acmg_class }} ({{ classif.acmg_translation }})</span></td>
							<td>
								<div class="w3-cell-row">
									<span class="w3-container w3-left-align w3-cell" id="comment_{{ classif.id }}-{{ classif.acmg_class }}-{{ variant_features.var_id }}">
									</span>
									{% autoescape false %}
										<script type="application/x-javascript">
											$("#comment_{{ classif.id }}-{{ classif.acmg_class }}-{{ variant_features.var_id }}").html(unescape('{{ classif.comment }}'));
                                        </script>
									{% endautoescape %}
									{% if classif.username == g.user['username'] %}
										{% if run_mode  and run_mode == 'on' %}
											<span class="w3-container w3-right-align w3-cell">
												<i class="fa fa-times w3-xlarge" style="color:red;cursor:pointer;" aria-hidden="true" title="click to remove this classification" onclick="remove_class('{{ variant_features.var_id }}', '{{ g.user['id'] }}', '{{ classif.acmg_class }}', '{{ url_for("ajax.remove_class") }}', '{{ csrf_token() }}')"></i>
											</span>
										{% endif %}
									{% endif %}
								</div>
							</td>
						</tr>
					{% endfor %}
				{% endif %}
				</tbody>
			</table><br/><br/>
			{% if g.user %}
				<div class="w3-container" id="message_return"></div>
				<div class="w3-row-padding">
					{% if run_mode  and run_mode == 'on' %}
						<div class="w3-col m4" id="modify_class">
							<span class="w3-button w3-blue w3-ripple w3-hover-light-blue w3-padding-16 w3-large" title="add your own classification to the history" style="width:100%" id="modify_class_btn" onclick="$('#modify_class_modal').show();">Add your classification</span>
							<div class="w3-modal" id="modify_class_modal" style="display:none;">
								<div class="w3-modal-content w3-display-middle w3-animate-opacity w3-mobile">
									<header class="w3-container w3-blue">
										<span class="w3-button w3-display-topright w3-xlarge" onclick="$('#modify_class_modal').hide();">X</span>
										<br />
										<p><span class="w3-xlarge">Add your ACMG classification to the history:</span></p>
										<br />
									</header>
									<div class="w3-container w3-center">
										<form method="post" class="w3-container w3-text-blue w3-margin w3-large" style="width:100%" id="acmg_form" enctype="application/x-www-form-urlencoded">
											 <!--action="{{ url_for('ajax.modif_class') }}"-->
											<input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
											<input type="hidden" name="variant_id" value="{{ variant_features.var_id }}" />
											<input type="hidden" name="mobiuser_id" value="{{ g.user['id'] }}" />
											<h3>This will be visible to all users:</h3><br />
											<div class="w3-row w3-section">
												<div class="w3-quarter w3-left-align">
													<label for="acmg_select" form="acmg_form">ACMG class:</label>
												</div>
												<div class="w3-rest">
													<select class="w3-select w3-border" name="acmg_select" id="acmg_select" form="acmg_form">
														<option value="1" style="color:#00A020;">Benign</option>
														<option value="2" style="color:#0000A0;">Likely Benign</option>
														<option value="3" style="color:#969696;">Uncertain Significance</option>
														<option value="4" style="color:#FF6020;">Likely Pathogenic</option>
														<option value="5" style="color:#FF0000;">Pathogenic</option>
													</select>
												</div><br /><br />
											</div>
											<div class="w3-row w3-section">
												<div class="w3-quarter w3-left-align">
													<label for="acmg_comment" form="acmg_form">Comments (optional, use english, please, limit 500 chars):</label>
												</div>
												<div class="w3-rest">
													<textarea name="acmg_comment" id="acmg_comment" rows="4" cols="50" maxlength="500" form="acmg_form"></textarea>
												</div>
											</div>
											<span class="w3-center">
												<input type="button" id = "sub" class="w3-button w3-section w3-blue w3-ripple"  onclick="modify_class('{{ variant_features.var_id }}', '{{ g.user['id'] }}', '{{ url_for("ajax.modif_class") }}', '{{ csrf_token() }}');" value="submit"/>
												<!--<input type="submit" id = "sub" class="w3-button w3-section w3-blue w3-ripple"  onclick="$('#html').css('cursor', 'progress');$('#acmg_form').css('cursor', 'progress');" value="submit"/>-->
											</span>
										</form>
									</div>
									<br /><br />
								</div>
							</div>
						</div>
					{% else %}
						<span>Please wait for the end of the maintenance to classify this variant.</span>
					{% endif %}
				</div>
				<div class="w3-modal" id="message_modal" style="display:none;">
					<div class="w3-modal-content w3-display-middle w3-animate-opacity w3-mobile">
						<header class="w3-container w3-blue">
							<span class="w3-button w3-display-topright w3-xlarge" onclick="$('#message_modal').hide();">X</span>
							<br />
							<p><span class="w3-xlarge">Use MobiDetails to contact a peer (<span id="receiver_username"></span>):</span></p>
							<br />
						</header>
						<div class="w3-container w3-center">
							<form method="post" class="w3-container w3-text-blue w3-margin w3-large" style="width:100%" id="message_form" enctype="application/x-www-form-urlencoded">
								<input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
								<input type="hidden" name="sender_id" id="sender_id" value="{{ g.user['id'] }}" />
								<input type="hidden" name="receiver_id" id="receiver_id" value="1" />
								<!--<input type="hidden" name="variant_mes" id="variant_mes" value="{{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }}" />-->
								<span>This message will be sent via MobiDetails. The contact will also receive your email address and be able to answer directly.</span><br />
								<p><span id="message_object">Object: Query via MobiDetails from {{ g.user['username'] }} concerning {{ variant_features.gene_name[1] }}.{{ variant_features.nm_version }}:c.{{ variant_features.c_name }}</span></p>
								<div class="w3-row w3-section w3-responsive">
									<div class="w3-quarter w3-left-align">
										<label for="message_body" form="message_form"><span >Type your message here:</span></label>
									</div>
									<div class="w3-rest">
										<textarea name="message_body" id="message_body" rows="7" cols="50" maxlength="1000" form="message_form"></textarea>
									</div>
								</div>
								<span class="w3-center">
									<input type="button" id="sub_message" class="w3-button w3-section w3-blue w3-ripple"  onclick="send_var_message('{{ url_for("ajax.send_var_message") }}', '{{ csrf_token() }}');" value="submit"/>
								</span>
							</form>
						</div>
						<br /><br />
					</div>
				</div>
			{% else %}
				<div class="w3-margin w3-panel w3-sand w3-leftbar w3-display-container">
					<p>To modify the classification history, you must be <a href='{{ url_for('auth.login') }}'>logged in</a> (you might want to <a href='{{ url_for('auth.register') }}' target='_blank'>create an account</a> before).<br />
						<!--<strong>This will also allow data sharing with <a href="http://www.lovd.nl" target="_blank">LOVD</a> (read <a href='{{ url_for("md.about") }}#data_sharing' target="_blank">more</a>).</strong>-->
					</p>
				</div>
			{% endif %}
		</div>
		<br/><br/>
		<!--eighth section-->
		<hr />
		{{ section_title('admin', 'Administrative information') }}
		<div class="w3-responsive w3-container">
			<table class="w3-table w3-centered display compact" style="width:90%" id="admin_table">
				<thead>
					<tr>
						<th class="w3-left-align">Features</th>
						<th class="w3-left-align">Values</th>
						<th class="w3-left-align">Descriptions</th>
					</tr>
				</thead>
				<tbody>
					<tr><!--creation user-->
						<td class="w3-left-align">Creation user:</td>
						{% if variant_features.username != 'mobidetails' and g.user %}
							<td class="w3-left-align"><a href="{{ url_for('auth.profile', mobiuser_id=variant_features.mobiuser_id) }}" target="_blank" title="Click to see {{ variant_features.username }} profile"><span id="owner_username">{{ variant_features.username }}</span></a></td>
						{% else %}
							<td class="w3-left-align"><span id="owner_username">{{ variant_features.username }}</span></td>
						{% endif %}
						<td class="w3-left-align"><em class="w3-small">User who created the variant</em></td>
					</tr>
					<tr><!--creation date-->
						<td class="w3-left-align">Creation date:</td>
						<td class="w3-left-align">{{ variant_features.creation_date }}</td>
						<td class="w3-left-align"><em class="w3-small">Date of creation in MD</em></td>
					</tr>
				</tbody>
			</table><br /><br />
			{% if g.user %}
				<span id="current_user" style="display:none;">{{ g.user['username'] }}</span>
			{% else %}
				<span id="current_user" style="display:none;">mobidetails</span>
			{% endif %}
		</div>
		<br/><br/>
<!--		<div class="w3-responsive w3-container" id="res_versions" style="display:none">
			<table class="w3-table w3-centered display compact" style="width:90%" id="resource_table">
				<caption>Links and versions of external resources used in MD:</caption>
				<thead>
					<tr>
						<th class="w3-left-align">Resource</th>
						<th class="w3-left-align">Version</th>
					</tr>
				</thead>
				<tbody>
					{% for resource in local_files %}
						{% if local_files[resource]['name'] != 'Human genome sequence' %}
							<tr>
								<td class="w3-left-align">
									<a href="{{ urls[local_files[resource]['short_name']] }}" target="_blank" title="Visit {{ local_files[resource]['name'] }}">{{ local_files[resource]['name'] }}</a>
								</td>
								<td class="w3-left-align">{{ local_files[resource]['version'] }}</td>
							</tr>
						{% endif %}
					{% endfor %}
				</tbody>
			</table>
		</div>-->
		<div class="w3-container w3-padding-64 w3-center" id="res_versions" style="display:none">
			<div class="w3-responsive w3-container">
				<table class="w3-table w3-centered display compact" style="width:90%" id="resource_table">
					<caption>References and versions of external resources used in MD:</caption>
					<thead>
						<tr>
							<th class="w3-left-align">Resource</th>
							<th class="w3-left-align">Version</th>
							<th class="w3-left-align">Reference</th>
						</tr>
					</thead>
					<tbody>
						{% for tool in external_tools %}
							<tr>
								<td class="w3-left-align">
									<a href="{{ urls[external_tools[tool]['short_name']] }}" target="_blank" rel="noopener noreferrer" title="Visit {{ tool }}">{{ tool }}</a>
								</td>
								<td class="w3-left-align">{{ external_tools[tool]['version'] }}</td>
								<td class="w3-left-align"><a href="{{ external_tools[tool]['paper'] }}" target="_blank" rel="noopener noreferrer" title="Paper">{{ external_tools[tool]['firstauthor'] }} et al., {{ external_tools[tool]['year'] }}</a></td>
							</tr>
						{% endfor %}
					</tbody>
				</table>
			</div>
		</div>
	</div>
	<script>
		$(document).ready(function() {
			litvar('{{ url_for('ajax.litvar') }}', '{{ csrf_token() }}');
			lovd('{{ url_for('ajax.lovd') }}', '{{ csrf_token() }}');
			intervar('{{ url_for('ajax.intervar') }}', '{{ csrf_token() }}');
			run_spip('{{ url_for('ajax.spip') }}', '{{ csrf_token() }}');
		});
	</script>
{% endblock %}
